### PR TITLE
Refactor record tests

### PR DESCRIPTION
**Pull request by OrleneMitchell**

### CODE REVIEW

**Review by x10an14-nav — CHANGES_REQUESTED** [review]
Small nitpick, as discussed on zoom =)
Looks very good! Nothing else to comment on, this was a clean PR!

File: core/src/test/java/nav/statusplattform/core/repositories/RecordRepositoryTest.java [hunk 1]
```diff
@@ -290,4 +290,12 @@ private RecordEntity getRandomizedRecordEntity() {
                 .setResponsetime(SampleData.getRandomResponseTime());
     }
 
+    private List<RecordEntity> generateRandomizedRecordEntities(ServiceEntity serviceEntity, int amount) {
+        List<RecordEntity> records = new ArrayList<>();
+        for (int i = 0; i < amount; i++) {
```
Comment (x10an14-nav) [inline core/src/test/java/nav/statusplattform/core/repositories/RecordRepositoryTest.java:295]:
```suggestion
    private List<RecordEntity> generateRandomizedRecordEntities(ServiceEntity serviceEntity, int size) {
        if (size <= 0) {
            throw IllegalArgumentException("`size` argument refers to size of list, must be non-zero positive integer.");
        }
        List<RecordEntity> records = new ArrayList<>();
        for (int i = 0; i < size; i++) {
```

This edge case was not considered here. Not sure if suggested code compiles.

**Review by x10an14-nav — CHANGES_REQUESTED** [review]
Otherwise, I have no inputs! Fix that and it's good to go!

File: core/src/test/java/nav/statusplattform/core/repositories/RecordRepositoryTest.java [hunk 2]
```diff
@@ -290,4 +290,13 @@ private RecordEntity getRandomizedRecordEntity() {
...
     }
 
+    public static List<RecordEntity> getRandomizedRecordEntitiesForService(ServiceEntity serviceEntity) {
+        List<RecordEntity> records = new ArrayList<>();
+        int numberOfRecords = SampleData.randomNonZeroPositiveInteger(12);
+        for (int i = 0; i < numberOfRecords; i++) {
+            records.add(SampleData.getRandomizedRecordEntityForService(serviceEntity));
+        }
+        return records;
+    }
```
[the remark this answers is not in the document]
Comment (OrleneMitchell) [reply core/src/test/java/nav/statusplattform/core/repositories/RecordRepositoryTest.java]: Renamed to describe what class is actually doing; Additionally, Access modifier changed from public static because the method is used solely within the class; Finally, added argument so value may be altered depending on the test criteria.

File: core/src/test/java/nav/statusplattform/core/repositories/RecordRepositoryTest.java [hunk 3]
```diff
@@ -290,4 +290,15 @@ private RecordEntity getRandomizedRecordEntity() {
                 .setResponsetime(SampleData.getRandomResponseTime());
     }
 
+    private List<RecordEntity> generateRandomizedRecordEntities(ServiceEntity serviceEntity, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("argument amount refers to size of list, must be either non-zero or positive integer.");
```
Comment (x10an14-nav) [inline core/src/test/java/nav/statusplattform/core/repositories/RecordRepositoryTest.java:295]: ## Choose either
```suggestion
            throw new IllegalArgumentException("argument amount refers to size of list, must be a non-zero and positive integer.");
```

## Or
```suggestion
            throw new IllegalArgumentException("Argument `amount` refers to size of list, must be non-zero & positive integer.");
```

Comment (OrleneMitchell) [reply core/src/test/java/nav/statusplattform/core/repositories/RecordRepositoryTest.java]: Corrected misleading IllegalArgumentException message.

**Review by x10an14-nav — APPROVED** [review]
LGTM! =)
